<?xml version="1.0"?>
<launch>

    <!-- 启动rviz -->
    <arg name="rviz" default="true" />

    <!-- 设置地图的配置文件 -->
    <arg name="2dmap_file" default="cwk_map__20250918_155202.yaml" />
    <arg name="map_path" default="/home/unitree/HongTu/G1Nav2D/map_folder" />

    <!-- <arg name="2dmap_file" default="$(find pcd_file)/grid_map/pcd2pgm/bbb.yaml" /> -->
    <arg name="2dmap_frame" default="map"/>
    <arg name="2dmap_topic" default="map_2d"/>      <!-- 注意map_server的地图话题名不能为数字开头 -->

    <!-- 运行地图服务器，并且加载设置的地图-->
    <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_path)/$(arg 2dmap_file) /map:=$(arg 2dmap_topic)" >
        <param name="frame_id" value="$(arg 2dmap_frame)"/>
    </node>

    <!-- rviz display -->
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" />
    </group>

</launch>